dora/src/cpu/x64.rs

```diff
@@ -11,3 +13,27 @@ pub fn flush_icache(_: *const u8, _: usize) {
     // no flushing needed on x86_64, but emit compiler barrier
     compiler_fence(Ordering::SeqCst);
 }
+
+pub fn has_popcnt() -> bool {
+    return *HAS_POPCNT;
+}
+
+pub fn has_lzcnt() -> bool {
+    // Too conservative, but we currently lack an easy way to check support for ABM instructions:
+    // Revisit after https://github.com/gz/rust-cpuid/issues/30 is resolved.
+    return *HAS_TZCNT;
+}
+
+pub fn has_tzcnt() -> bool {
+    return *HAS_TZCNT;
+}
+
+lazy_static! {
+static ref FEATURES: FeatureInfo = CpuId::new().get_feature_info().unwrap();
+static ref FEATURES_EXTENDED: ExtendedFeatures = CpuId::new().get_extended_feature_info().unwrap();
+
+static ref HAS_POPCNT: bool = FEATURES.has_popcnt();
+static ref HAS_TZCNT: bool = FEATURES_EXTENDED.has_bmi1();
+// support for MULX, RORX, SARX, SHRX, SHLX
+static ref HAS_X_OPS: bool = FEATURES_EXTENDED.has_bmi2();
```

Comment:
This field has no `pub` function, because both the "bad old" instructions as well as the possible "better" replacements are intrinsic, so we don't need to make this decision from the outside.